json/src/value.rs

```diff
@@ -473,6 +559,99 @@ impl de::Deserialize for Value {
     }
 }
 
+/// Represents type of a JSON value
+#[derive(Clone, Copy, Eq, PartialEq, Debug)]
+pub enum ValueType {
```

Comment:
This type sounds a lot like [`std::mem::discriminant_value`](https://github.com/rust-lang/rust/pull/34785), so we could just wait until that is stable (3 Months probably)